mentat/code_feature.py

```diff
@@ -156,36 +158,35 @@ def contains_line(self, line_number: int):
     def _get_code_message(self) -> list[str]:
         session_context = SESSION_CONTEXT.get()
         code_file_manager = session_context.code_file_manager
-        git_root = session_context.git_root
         parser = session_context.config.parser
 
         code_message: list[str] = []
 
         # We always want to give GPT posix paths
-        abs_path = Path(git_root / self.path)
-        rel_path = Path(os.path.relpath(abs_path, git_root))
-        posix_rel_path = Path(rel_path).as_posix()
-        filename = f"{posix_rel_path}"
-        code_message.append(filename)
+        if self.path.is_relative_to(session_context.cwd):
+            code_message_path = self.path.relative_to(session_context.cwd).as_posix()
+        else:
+            code_message_path = self.path.as_posix()
+        code_message.append(str(code_message_path))
```

Comment:
I'm not sure if this will trip up edit quality from LLMs if we have both relative and absolute paths mixed in. If we wan to keep using relative paths for files outside of the cwd, we can set `walk_up` to `True` (see the [python docs](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to) for this) and on any errors just default to absolute paths

Reply:
Hmm, yeah I don't think we want to give the LLM absolute paths. I think setting walkup and on error going to absolute path is the best solution.

Reply:
So it turns out the `walk_up` argument is not available for python 3.10 (the earliest version we support), so I'll add in a function to do this